velox/common/memory/MmapAllocator.cpp

```diff
@@ -53,35 +53,41 @@ bool MmapAllocator::allocateNonContiguous(
 
   const SizeMix mix = allocationSize(numPages, minSizeClass);
 
-  const auto numFreed = freeInternal(out);
+  const int64_t numFreed = freeInternal(out);
   if (numFreed != 0) {
     numAllocated_.fetch_sub(numFreed);
   }
 
-  if (injectedFailure_ == Failure::kCap) {
-    if (!isPersistentFailureInjection_) {
-      injectedFailure_ = Failure::kNone;
+  if (testingInjectFailure(Failure::kCap)) {
+    if (reservationCB != nullptr) {
+      reservationCB(numFreed * AllocationTraits::kPageSize, false);
```

Comment:
nit: * AllocationTraits::kPageSize makes the code long and hard to read; I wonder if there is a way to shorten the code by using an inline helper function:

```
pageBytes(numFreed)
```

Reply:
I add a method in AllocationTraits for this.